src/Uno.Extensions.Reactive/Core/ListState.Extensions.cs

```diff
@@ -122,6 +122,60 @@ public static ValueTask AddAsync<T>(this IListState<T> state, T item, Cancellati
 	public static ValueTask RemoveAllAsync<T>(this IListState<T> state, Predicate<T> match, CancellationToken ct = default)
 		=> state.UpdateDataAsync(itemsOpt => itemsOpt.Map(items => items.RemoveAll(match)), ct);
 
+
+
+	/// <summary>
+	/// Updates all items from a list state that match the key of <paramref name="oldT"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of the items in the list.</typeparam>
+	/// <param name="state">The list state onto which the item should be added.</param>
+	/// <param name="oldT">The old value of the item.</param>
+	/// <param name="newT">The new value for the item.</param>
+	/// <param name="ct">A token to abort the async add operation.</param>
+	/// <returns></returns>
+	public static ValueTask UpdateItemAsync<T>(this IListState<T> state, T oldT,  T newT, CancellationToken ct = default) where T : IKeyEquatable<T>
```

Comment:
To be consistent with other methods like this (e.g. [`State.SetAsync`](https://github.com/unoplatform/uno.extensions/blob/5f3c45f398fbfee9357404d90ff060ba2169bfc2/src/Uno.Extensions.Reactive/Core/State.Extensions.cs#L125)) in order to protect ACID properties, I would add a constraint on `T` : `struct` (and add overloads with `T?` and `string`).

Also not a name of the parameters, I would rename those:
```suggestion
	public static ValueTask UpdateItemAsync<T>(this IListState<T> state, T oldItem,  T newItem, CancellationToken ct = default) where T : IKeyEquatable<T>
```

Reply:
@dr1rrb  Added a new overload and also updated the parameter names.